Makefile

```diff
@@ -1,7 +1,7 @@
 # Build, package, test, and clean
 PROJECT=pygmt
 TESTDIR=tmp-test-dir-with-unique-name
-PYTEST_ARGS=--cov-config=../.coveragerc --cov-report=term-missing --cov=$(PROJECT) --doctest-modules -v --mpl --mpl-results-path=results --pyargs
+PYTEST_ARGS=--cov-config=../.coveragerc --cov-report=term-missing --cov=$(PROJECT) --doctest-modules -v --mpl --mpl-results-path=results --pyargs ${PYTEST_EXTRA}
```

Comment:
This won't work, because we didn't declare the `$PYTEST_EXTRA` environment variable. Also need to remove the `$PYTEST_EXTRA` string from the ci_tests.yaml file.

Reply:
I think it works. It doesn't have to be an environmental variable. `make test` and `make test PYTEST_EXTRA="-r P"` are using different pytext arguments.

Reply:
Now although the tests all pass, we actually can see more "errors" by using `-r P`:
```
______________________ test_call_module_invalid_arguments ______________________
----------------------------- Captured stderr call -----------------------------
gmtinfo [ERROR]: Error for input file: No such file (bogus-data.bla)
________________________ test_call_module_error_message ________________________
----------------------------- Captured stderr call -----------------------------
gmtinfo [ERROR]: Error for input file: No such file (bogus-data.bla)
__________________________ test_extract_region_fails ___________________________
----------------------------- Captured stderr call -----------------------------
pygmt-session [ERROR]: No hidden PS file found
____________________________ test_get_default_fails ____________________________
----------------------------- Captured stderr call -----------------------------
pygmt-session [ERROR]: Syntax error: Unrecognized keyword NOT_A_VALID_NAME
________________________ test_text_nonexistent_filename ________________________
----------------------------- Captured stderr call -----------------------------
text [ERROR]: Error for input file: No such file (notexist.txt)
_______________________________ test_which_fails _______________________________
----------------------------- Captured stderr call -----------------------------
gmtwhich [ERROR]: File 5a9f5f5cb31c4bb99525d0d82eaf0fc6 not found!

```

Reply:
Isn't it because the CI (Travis and GH Actions) still uses `make test PYTEST_EXTRA="-r P"`? If we run `make test` locally after checking out this branch, it probably won't work.

Reply:
Yes. Look at the captured stderr above:
```
gmtinfo [ERROR]: Error for input file: No such file (bogus-data.bla)
```
This is a GMT error message, but I believe it's already corrected captured in the pygmt test.

I believe the initial idea of PYTEST_EXTRA is that, when we or other users run `make test` locally, we won't see these "error" messages, as they're a little confusing. But in CI, `make test PYTEXT_EXTRA="-r P"` gives more verbose messages, useful for debugging.

Reply:
Then we don't actually need to change the Makefile. You'll see that `PROJECT` is declared before using `$(PROJECT)`, but we don't declare `PYTEST_EXTRA` before using `$(PYTEST_EXTRA)`, it is effectively blank.

Reply:
If you run `make test`, then `PYTEST_EXTRA` is blank; if you run `make test PYTEST_EXTRA="-r P"`, then variable PYTEST_EXTRA is defined in shell, and passed into Makefile.

Reply:
`make test` runs:
```
cd tmp-test-dir-with-unique-name; pytest --cov-config=../.coveragerc --cov-report=term-missing --cov=pygmt --doctest-modules -v --mpl --mpl-results-path=results --pyargs  pygmt
```
and `make test PYTEST_EXTRA="-r P"` runs:
```
cd tmp-test-dir-with-unique-name; pytest --cov-config=../.coveragerc --cov-report=term-missing --cov=pygmt --doctest-modules -v --mpl --mpl-results-path=results --pyargs -r P pygmt
```

Reply:
Right, I think I get it now. It wasn't working properly before, but now it does.